### PR TITLE
Chore: Rename IcrcTransactionList to UiTransactionList

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Integrate ICP index canister.

In this PR, rename IcrcTransactionsList to UiTransactionsList and use directly the TransactionCard instead of IcrcTransactionCard.

# Changes

* Rename IcrcTransactionsList to UiTransactoinsList and use TransactionCard
* Remove IcrcTransactionCard.

# Tests

* Remove test for IcrcTransactionCard
* Rename test file.
* Rename IcrcTransactionsListPo and methods to get it.

# Todos

- [ ] Add entry to changelog (if necessary).

Not necessary.

### CODE REVIEW

**Comment by lmuntaner** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #4501** <a href="https://app.graphite.dev/github/pr/dfinity/nns-dapp/4501?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dfinity/nns-dapp/4501?utm_source=stack-comment).

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/tests/page-objects/CkBTCWallet.page-object.ts [hunk 1]
```diff
@@ -14,8 +14,8 @@ export class CkBTCWalletPo extends BasePageObject {
     return new CkBTCWalletPo(element.byTestId(CkBTCWalletPo.TID));
   }
 
-  getIcrcTransactionsListPo(): IcrcTransactionsListPo {
-    return IcrcTransactionsListPo.under(this.root);
+  getTransactionsListPo(): UiTransactionsListPo {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/CkBTCWallet.page-object.ts:17]: Can you name this `getUiTransactionsListPo`? I find it sometimes confusing when the name of the getter doesn't match the name of the PO.

File: frontend/src/tests/page-objects/SnsWallet.page-object.ts [hunk 2]
```diff
@@ -22,8 +22,8 @@ export class SnsWalletPo extends BasePageObject {
     return WalletPageHeadingPo.under(this.root);
   }
 
-  getIcrcTransactionsListPo(): IcrcTransactionsListPo {
-    return IcrcTransactionsListPo.under(this.root);
+  getTransactionsListPo(): UiTransactionsListPo {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/SnsWallet.page-object.ts:25]: same

File: frontend/src/tests/lib/components/accounts/UiTransactionsList.spec.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 import type { UiTransaction } from "$lib/types/transaction";
 import { mockIcrcTransactionsStoreSubscribe } from "$tests/mocks/icrc-transactions.mock";
 import { createMockUiTransaction } from "$tests/mocks/transaction.mock";
-import { IcrcTransactionsListPo } from "$tests/page-objects/IcrcTransactionsList.page-object";
+import { UiTransactionsListPo } from "$tests/page-objects/UiTransactionsList.page-object";
 import { JestPageObjectElement } from "$tests/page-objects/jest.page-object";
 import { render } from "@testing-library/svelte";
 
-describe("IcrcTransactionList", () => {
+describe("UiTransactionList", () => {
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/accounts/UiTransactionsList.spec.ts:10]: `"UiTransactionsList"` plural to match the name of the file

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/page-objects/CkBTCWallet.page-object.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/tests/page-objects/CkBTCWallet.page-object.ts]: done

File: frontend/src/tests/lib/components/accounts/UiTransactionsList.spec.ts (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/components/accounts/UiTransactionsList.spec.ts]: done

File: frontend/src/tests/page-objects/SnsWallet.page-object.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/page-objects/SnsWallet.page-object.ts]: done

**Pull request merged**